go/mysql/collations/collation.go

```diff
@@ -144,10 +144,10 @@ type Collation interface {
 	IsBinary() bool
 
 	// GetUpperTable returns the upper table about related encoding
-	ToUpper([]byte) []byte
+	ToUpper([]byte, []byte) []byte
```

Comment:
Like in the other APIs, the `dst` buffer should be the first argument for the function, not the last. You should also name the arguments and update the comments!